pkg/agent/clientset.go

```diff
@@ -179,16 +184,17 @@ func (cs *ClientSet) resetBackoff() *wait.Backoff {
 }
 
 // sync makes sure that #clients >= #proxy servers
-func (cs *ClientSet) sync() {
+func (cs *ClientSet) sync(ctx context.Context) {
```

Comment:
The lease system lets us make this function a lot smarter, but the changes here don't really take advantage of it. Are you planning to follow up with those changes?

Specifically, I'm thinking of:
1. When a new lease is added, agents should try to connect to it. This is related to the informers comment elsewhere. This also means we don't need to continuously poll any more.
2. clientset should ensure that there is a server connection for each valid lease. This could avoid some edge cases where the agent has the right number of connections, but isn't connected to the right servers. (I'm not sure how this could happen in practice, but maybe if the loadbalancer & KCP instances get out of sync).

Reply:
Will not have time to do this in this PR, but I will open issues for these two things!